test/integration_tests/test_create_index_executor.py

```diff
@@ -156,8 +163,42 @@ def test_should_cleanup_when_exception(self, faiss_mock):
         with self.assertRaises(ExecutorError):
             execute_query_fetch_all(query)
 
-        # Check secondary index is dropped.
-        secondary_index_tb_name = "secondary_index_{}_{}".format(
-            "HNSW", "testCreateIndexName"
+        # Check faulty index is not persisted on the disk
+        self.assertFalse(os.path.exists(self._index_save_path()))
+
+    def test_should_create_index_with_udf(self):
```

Comment:
Do we need another test without specifying `USING HNSW`?

Reply:
I think this will lead to some ambiguities. As @xzdandy suggested, I will throw an error for this case in the binder.